tools/mc-java-protoc/src/main/java/io/spine/tools/mc/java/protoc/field/FieldGen.java

```diff
@@ -73,27 +71,85 @@ private FieldGen(ImmutableList<CodeGenerationTask> tasks) {
     /**
      * Creates a new instance based on the passed Protoc config.
      */
-    @SuppressWarnings("MethodWithMultipleLoops") // Required to configure code generation tasks.
     public static FieldGen instance(SpineProtocConfig spineProtocConfig) {
         checkNotNull(spineProtocConfig);
-        AddFields config = spineProtocConfig.getAddFields();
-
         ImmutableList.Builder<CodeGenerationTask> tasks = ImmutableList.builder();
-        EntityStateConfig entityStateConfig = config.getEntityStateConfig();
-        if (isNotDefault(entityStateConfig)) {
-            tasks.add(new GenerateEntityStateFields(entityStateConfig, factory));
+        addFromEntities(tasks, spineProtocConfig);
```

Comment:
Please introduce a helper class which would accept `spineProtocConfig` on constructor, and create `tasks` in this constructor. Under this class you'll be be able to gather de-staticized versions of the below methods. As instance methods they won't need such lengthy parameters.